app/main/views/suppliers.py

```diff
@@ -107,16 +110,36 @@ def supplier_details():
     supplier['contact'] = supplier['contactInformation'][0]
     country_name = get_country_name_from_country_code(supplier.get('registrationCountry'))
 
-    framework = None
+    supplier_company_details_confirmed = supplier['companyDetailsConfirmed']
+    application_company_details_confirmed = None
+
+    frameworks = data_api_client.find_frameworks().get('frameworks')
+    framework_slug_map = {framework['slug']: framework for framework in frameworks}
+    open_framework_slugs = [framework['slug'] for framework in frameworks if framework['status'] == 'open']
+
+    unconfirmed_open_supplier_frameworks, unconfirmed_open_supplier_framework_names = [], []
+    if open_framework_slugs:
+        unconfirmed_open_supplier_frameworks = list(filter(
+            lambda sf: sf['frameworkSlug'] in open_framework_slugs and not sf['applicationCompanyDetailsConfirmed'],
+            data_api_client.get_supplier_frameworks(supplier_id=current_user.supplier_id)['frameworkInterest']
+        ))
+        unconfirmed_open_supplier_framework_names = [
+            framework_slug_map[sf['frameworkSlug']]['name'] for sf in unconfirmed_open_supplier_frameworks
+        ]
+
+    currently_applying_to_framework = None
+    if "currently_applying_to" in session and session["currently_applying_to"] in framework_slug_map:
```

Comment:
Is the second conditional here to make sure the user hasn't snuck a dodgy framework slug into their session somehow?

Reply:
That should be impossible (without discovering our secret key). It's more just a belt and braces approach for if we somehow set an invalid framework slug. It's 99% unnecessary, but given how simple a check it is, still vaguely worth doing.